packages/plugin-hardhat/src/upgrade-proxy.ts

```diff
@@ -46,9 +46,9 @@ export function makeUpgradeProxy(hre: HardhatRuntimeEnvironment, platformModule:
     const adminBytecode = await getCode(provider, adminAddress);
 
     if (isEmptySlot(adminAddress) || adminBytecode === '0x') {
-      // No admin contract: use TransparentUpgradeableProxyFactory to get proxiable interface
-      const TransparentUpgradeableProxyFactory = await getTransparentUpgradeableProxyFactory(hre, signer);
-      const proxy = TransparentUpgradeableProxyFactory.attach(proxyAddress);
+      // No admin contract: use ITransparentUpgradeableProxyFactory to get proxiable interface
+      const ITransparentUpgradeableProxyFactory = await getITransparentUpgradeableProxyFactory(hre, signer);
```

Comment:
Oh, I remember now, and I understand why this was done.

Reply:
Commenting for completeness:

The `upgradeProxy` function in the plugins invokes the `upgradeToAndCall` or `upgradeTo` function on the proxy, regardless of whether it is a UUPS or Transparent proxy.  It did this by using TransparentUpgradeableProxy's contract factory to encode those functions, which depends on the ABI.  (Note this also works for UUPS proxies because those function signatures are the same between UUPS and Transparent).

In OpenZeppelin Contracts 4.8.3, the `upgradeToAndCall` and `upgradeTo` functions are not part of TransparentUpgradeableProxy's ABI, but are in the interface [ITransparentUpgradeableProxy's ABI](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.8.3/contracts/proxy/transparent/TransparentUpgradeableProxy.sol#L14).  Therefore, we use ITransparentUpgradeableProxy's contract factory to encode those functions instead.